client/site-performance/components/PageSelector.tsx

```diff
@@ -0,0 +1,92 @@
+import page from '@automattic/calypso-router';
+import { SearchableDropdown } from '@automattic/components';
+import { useDebouncedInput } from '@wordpress/compose';
+import { useI18n } from '@wordpress/react-i18n';
+import { useMemo } from 'react';
+import QueryPosts from 'calypso/components/data/query-posts';
+import { useSelector } from 'calypso/state';
+import { getPostsForQueryIgnoringPage } from 'calypso/state/posts/selectors';
+import getCurrentQueryArguments from 'calypso/state/selectors/get-current-query-arguments';
+import { getSelectedSite } from 'calypso/state/ui/selectors';
+
+interface Page {
+	slug: string;
+	URL: string;
+	title: string;
+}
+
+const PER_PAGE = 10;
+
+export const PageSelector = () => {
+	const { __ } = useI18n();
+
+	const queryParams = useSelector( getCurrentQueryArguments );
+	const site = useSelector( getSelectedSite );
+
+	const [ , setPageQuery, pageQuery ] = useDebouncedInput();
+	const query = { type: 'page', number: PER_PAGE, search: pageQuery, page: 1 };
+	const pages = useSelector( ( state ): Page[] | null =>
```

Comment:
I'm unsure if we _actually_ want to nail this down to pages. I believe we want to fetch all possible pages from the blog, including posts, categories, etc. Is that correct?